lib/util/sdp/index.js

```diff
@@ -251,6 +289,84 @@ function setSimulcast(sdp, sdpFormat, trackIdsToAttributes) {
   })).concat('').join('\r\n');
 }
 
+/**
+ * Filter local codecs based on the remote unified plan SDP.
+ * @param {string} localSdp
+ * @param {string} remoteSdp
+ * @returns {string} - Updated local SDP
+ */
+function unifiedPlanFilterLocalCodecs(localSdp, remoteSdp) {
+  const localMediaSections = getMediaSections(localSdp);
+  const localSession = localSdp.split('\r\nm=')[0];
+  const remoteMidsToMediaSections = createMidToMediaSectionMap(remoteSdp);
+  return [localSession].concat(localMediaSections.map(localSection => {
+    const localMid = getMidForMediaSection(localSection);
+    const remoteSection = localMid && remoteMidsToMediaSections.get(localMid);
+    // Do nothing if either the local m= section does not have an equivalent
+    // remote m= section or if it is neither an audio nor a video m= section.
+    if (!(remoteSection && /^m=(audio|video)/.test(localSection))) {
+      return localSection;
+    }
+    // Construct a Map of the remote Payload Types to their codec names.
+    const remotePtToCodecs = createPtToCodecName(remoteSection);
+    // Construct a Map of the local codec names to their Payload Types.
+    const localCodecToPts = createCodecMapForMediaSection(localSection);
+    // Maintain a list of local Payload Types to retain.
+    let localPts = [];
+
+    remotePtToCodecs.forEach((remoteCodec, remotePt) => {
+      const matchingLocalPts = localCodecToPts.get(remoteCodec) || [];
+      // If there is only one local Payload Type that matches the remote codec,
+      // retain it.
+      if (matchingLocalPts.length === 1) {
```

Comment:
not really sure if its a case to be considered, but maybe leave early here?
```
if (matchingLocalPts.length === 0) { 
   return; 
}
```

Reply:
@makarandp0 I think if the length is 0, then the 2 `if` blocks are skipped and we have the desired effect that you have mentioned.